internal/query/projection/group_users.go

```diff
@@ -0,0 +1,165 @@
+			handler.NewColumn(GroupUsersColumnResourceOwner, handler.ColumnTypeText),
+			handler.NewColumn(GroupUsersColumnInstanceID, handler.ColumnTypeText),
+			handler.NewColumn(GroupUsersColumnSequence, handler.ColumnTypeInt64),
+			handler.NewColumn(GroupUsersColumnCreationDate, handler.ColumnTypeTimestamp),
+		},
+			handler.NewPrimaryKey(GroupUsersColumnInstanceID, GroupUsersColumnGroupID, GroupUsersColumnUserID),
+			handler.WithIndex(handler.NewIndex("user_id", []string{GroupUsersColumnUserID})),
+			handler.WithIndex(handler.NewIndex("group_id", []string{GroupUsersColumnGroupID})),
+		),
+	)
+}
+
+func (g *groupUsersProjection) Reducers() []handler.AggregateReducer {
+	return []handler.AggregateReducer{
+		{
+			Aggregate: group.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  group.GroupUsersAddedEventType,
+					Reduce: g.reduceGroupUsersAdded,
+				},
+				{
+					Event:  group.GroupUsersRemovedEventType,
+					Reduce: g.reduceGroupUsersRemoved,
+				},
+			},
+		},
+		{
+			Aggregate: group.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  group.GroupRemovedEventType,
+					Reduce: g.reduceGroupRemoved,
+				},
+			},
+		},
+		{
+			Aggregate: org.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  org.OrgRemovedEventType,
+					Reduce: g.reduceOwnerRemoved,
+				},
+			},
+		},
+		{
+			Aggregate: instance.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  instance.InstanceRemovedEventType,
+					Reduce: reduceInstanceRemovedHelper(GroupColumnInstanceID),
+				},
+			},
+		},
+	}
+}
+
+func (g *groupUsersProjection) reduceGroupUsersAdded(event eventstore.Event) (*handler.Statement, error) {
+	e, err := assertEvent[*group.GroupUsersAddedEvent](event)
+	if err != nil {
+		return nil, err
+	}
+
+	stmts := make([]func(eventstore.Event) handler.Exec, 0, len(e.UserIDs))
+	for _, userID := range e.UserIDs {
+		stmts = append(stmts, handler.AddCreateStatement(
+			[]handler.Column{
+				handler.NewCol(GroupUsersColumnGroupID, e.Aggregate().ID),
+				handler.NewCol(GroupUsersColumnUserID, userID),
+				handler.NewCol(GroupUsersColumnResourceOwner, e.Aggregate().ResourceOwner),
+				handler.NewCol(GroupUsersColumnInstanceID, e.Aggregate().InstanceID),
+				handler.NewCol(GroupUsersColumnSequence, e.Sequence()),
+				handler.NewCol(GroupUsersColumnCreationDate, e.CreationDate()),
+			},
+		))
+	}
+	return handler.NewMultiStatement(e, stmts...), nil
+}
+
+func (g *groupUsersProjection) reduceGroupUsersRemoved(event eventstore.Event) (*handler.Statement, error) {
+	e, err := assertEvent[*group.GroupUsersRemovedEvent](event)
+	if err != nil {
+		return nil, err
+	}
+	stmts := make([]func(eventstore.Event) handler.Exec, 0, len(e.UserIDs))
+	for _, userID := range e.UserIDs {
+		stmts = append(stmts, handler.AddDeleteStatement(
+			[]handler.Condition{
+				handler.NewCond(GroupUsersColumnGroupID, e.Aggregate().ID),
```

Comment:
Group is under organization, so if you remove the group without organization as conditional there is the possibility that you remove the group under a different organization as well.